core/templates/services/playthrough.service.ts

```diff
@@ -17,383 +17,268 @@
     }
+    this.visitedStates = [destStateName];
+  }
+}
 
-    private storePlaythrough(isNewPlaythrough: boolean): void {
-      let playthroughId = (
-        isNewPlaythrough ? null : PlaythroughService.playthrough.playthroughId);
-      let promise = this.http.post(this.getFullPlaythroughUrl(), {
-        playthrough_data: PlaythroughService.playthrough.toBackendDict(),
-        issue_schema_version: ServicesConstants.CURRENT_ISSUE_SCHEMA_VERSION,
-        playthrough_id: playthroughId
-      }).toPromise();
-      if (isNewPlaythrough) {
-        promise.then((
-            response: {
-                // eslint-disable-next-line camelcase
-                playthrough_stored: boolean, playthrough_id: string }) => {
-          if (response.playthrough_stored) {
-            // In cases where maximum number of playthroughs already exists, the
-            // above flag is not True and playthrough ID is not set.
-            PlaythroughService.playthrough.playthroughId = (
-              response.playthrough_id);
-          }
-        });
-      }
-    }
+class EarlyQuitTracker {
+  constructor(
+      public stateName: string = null,
+      public timeSpentInStateSecs: number = null) {}
 
-    private getFullPlaythroughUrl(): string {
-      return this.urlInterpolationService.interpolateUrl(
-        ServicesConstants.STORE_PLAYTHROUGH_URL, {
-          exploration_id: PlaythroughService.playthrough.expId
-        });
-    }
+  isIssue(): boolean {
+    return (
+      this.timeSpentInStateSecs !== null &&
+      this.timeSpentInStateSecs < ServicesConstants.EARLY_QUIT_THRESHOLD_IN_SECS
+    );
+  }
 
-    private isPlaythroughDiscarded(): boolean {
-      return !this.explorationFeaturesService.isPlaythroughRecordingEnabled() ||
-        !PlaythroughService.isLearnerInSamplePopulation;
-    }
+  recordExplorationQuit(stateName: string, timeSpentInStateSecs: number): void {
+    this.stateName = stateName;
+    this.timeSpentInStateSecs = timeSpentInStateSecs;
+  }
+}
 
-    initSession(
-        explorationId: string, explorationVersion: number,
-        playthroughProbability: number): void {
-      PlaythroughService.isLearnerInSamplePopulation =
-        this.determineIfLearnerIsInSamplePopulation(playthroughProbability);
-      PlaythroughService.playthrough = this.playthroughObjectFactory.createNew(
-        null, explorationId, explorationVersion, null, {}, []);
-      PlaythroughService.expStopwatch = this.stopwatchObjectFactory.create();
-      PlaythroughService.multipleIncorrectStateName = {
-        state_name: null,
-        num_times_incorrect: null
-      };
-      PlaythroughService.cycleIdentifier = {
-        cycle: null,
-        num_cycles: null
-      };
-      PlaythroughService.visitedStates = [];
-      PlaythroughService.misTracker = false;
-      PlaythroughService.cstTracker = false;
-    }
+@Injectable({
+  providedIn: 'root'
+})
+export class PlaythroughService {
+  private explorationId?: string = null;
```

Comment:
Do you need all the question marks? I checked e.g. state-editor.service.ts and it doesn't seem to have it.